feat: Add event tracking for table quickstart experiment

seanoliver opened the pull request:
## Summary

Implements telemetry instrumentation for the table quickstart experiment to measure exposure and engagement across AI, Templates, and Assistant variations.

## Changes

### Event Definitions (7 new events)
- `table_quickstart_viewed` - Exposure tracking for variant assignment
- `table_quickstart_ai_prompt_submitted` - AI prompt submissions
- `table_quickstart_ai_generation_completed` - AI generation success/failure
- `table_quickstart_template_selected` - Template selection (unified for both AI-generated and pre-made templates)
- `table_quickstart_quick_idea_clicked` - Quick idea button clicks
- `table_quickstart_category_selected` - Template category navigation
- `table_quickstart_assistant_opened` - Assistant chat creation

Note: Conversion tracking is intentionally omitted. We rely on joining `table_quickstart_viewed` with existing `table_created` events in post-analysis rather than creating experiment-specific conversion events.

### Implementation
- Added tracking to `QuickstartAIWidget.tsx` (4 events)
- Added tracking to `QuickstartTemplatesWidget.tsx` (2 events)
- Added tracking to `NewTab.tsx` (2 events: exposure + assistant)

## Test Plan

- [x] Verify events fire in PostHog staging
- [x] Test all three variations (AI, Templates, Assistant)
- [x] Confirm variant attribution in exposure events
- [x] Validate interaction events fire correctly
- [x] Verify unified template selection event captures source correctly

## Metrics Enabled

This PR enables measurement of:
- Variant exposure rates
- Engagement by variation (AI vs Templates vs Assistant)
- Template selection patterns (unified across AI and template variations)
- AI generation success rates
- Conversion rates via join with `table_created` event
- Drop-off points in each flow

## Related

- Follows PR #39745 (type-safe event tracking utility)
- Complements PR #39825 (assistant variation implementation)
- Part of table quickstart experiment (PostHog flag: `tableQuickstart`)

seanoliver committed:
Add event tracking for table quickstart experiment

Implements telemetry instrumentation for the table quickstart experiment
to measure exposure and engagement across AI, Templates, and Assistant variations.

Events added:
- table_quickstart_viewed: Exposure tracking for variant assignment
- table_quickstart_ai_prompt_submitted: AI prompt submissions
- table_quickstart_ai_generation_completed: AI generation success/failure
- table_quickstart_template_selected: Template selection (unified for AI and templates)
- table_quickstart_quick_idea_clicked: Quick idea button clicks
- table_quickstart_category_selected: Template category navigation
- table_quickstart_assistant_opened: Assistant chat creation

All events use type-safe useTrack() hook with automatic project/org context.
Conversion tracking intentionally omitted - relying on joins with existing
table_created events for post-analysis.

pamelachia commented on packages/common/telemetry-constants.ts:
```diff
@@ -2134,6 +2287,13 @@ export type TelemetryEvent =
+  | TableQuickstartAIGenerationCompletedEvent
+  | TableQuickstartQuickIdeaClickedEvent
+  | TableQuickstartCategorySelectedEvent
+  | TableQuickstartTemplateSelectedEvent
```
same with this

seanoliver committed:
event name cleanup

awaseem reviewed (commented):
@seanoliver might be a dumb question, but how can I test the AI variant changes? they seem to behind a flag

seanoliver committed:
Address code review feedback

- Use nullish coalescing operator (??) instead of logical OR (||) for categoryName fallback
- Refactor handleGenerateTables to accept object parameter for improved call site readability

awaseem reviewed (approved):
Tested variants in Posthog, events seem to be coming through in Staging. LGTM nice work @seanoliver

seanoliver committed:
prettier

seanoliver committed:
action key name alignment with event name

seanoliver committed:
Merge branch 'master' into feat/table-quickstart-tracking-v2
